src/coreclr/jit/compiler.h

```diff
@@ -1930,8 +1930,9 @@ struct NaturalLoopIterInfo
 //   loop can reach every other block of the loop.
 //
 // * All loop blocks are dominated by the header block, i.e. the header block
-//   is guaranteed to be entered on every iteration. Note that in the prescence
-//   of exceptional flow the header might not fully execute on every iteration.
+//   is guaranteed to be entered on every iteration. Note that in the presence
+//   of exceptional flow the header might not fully execute on every iteration;
+//   however, after loop canonicalization we do have this property.
```

Comment:
This confused me a little until I looked at the code. I think you are saying that the code in the header either fully executes or on exception exits the loop entirely (so there are no more iterations if an exception happens).

Reply:
Hmm, yes, that is what I meant, but I went back to your canonicalization PR and we already discussed and found that it is not true because of a special case: https://github.com/dotnet/runtime/pull/110880#issuecomment-2576357388

So this change should be reverted.